### PR TITLE
Initialize v2.0.0 for jwt behavioral change.

### PR DESCRIPTION
**Pull request by mpmadhavig**
Related issues:
- https://github.com/wso2/product-is/issues/20880

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 1]
```diff
@@ -1238,18 +1238,24 @@ public static String getApplicationUpdatedVersion(ServiceProvider serviceProvide
...
 
-        // Since there will be new versions onboarded in the future, initialized this as a switch not if.
-        if (currentVersion.equals(ApplicationConstants.ApplicationVersion.APP_VERSION_V0)) {
-            if (!inboundConfigType.equals(IdentityApplicationConstants.OAuth2.NAME)) {
-                currentVersion = ApplicationConstants.ApplicationVersion.APP_VERSION_V1;
-            }
+        switch (currentVersion) {
+            case ApplicationConstants.ApplicationVersion.APP_VERSION_V0:
+            case ApplicationConstants.ApplicationVersion.APP_VERSION_V1:
+                if (inboundConfigType != null && !inboundConfigType.equals(IdentityApplicationConstants.OAuth2.NAME)) {
```
Comment (DMHP) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1244]: Swap the constant with the variable

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 2]
```diff
@@ -1238,18 +1238,24 @@ public static String getApplicationUpdatedVersion(ServiceProvider serviceProvide
...
+                    currentVersion = ApplicationConstants.ApplicationVersion.APP_VERSION_V2;
```
Comment (DMHP) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1245]: Is this logic correct? If someone has already upgraded his application to V1 after modifying the application logic, are we automatically upgrading the version of those applications to v2?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java (hunk 1 above)
Comment (mpmadhavig) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java]: The null check was there for apps with no inbound auth method configured. But we can change that logic to behave as same as non-oidc app. There for changing the code as below.

834a75d5812a461ce8d80e2386d84292e986fbb7

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java (hunk 2 above)
Comment (mpmadhavig) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1245]: No. The logic here is, if the app version is either v0 or v1 AND the app inbound auth type is not OIDC then automatically upgrade the version to v2.

**Review by DMHP — APPROVED** [review]

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**